### PR TITLE
Add cloud credential validation

### PR DESCRIPTION
**Pull request by mjura**

### CODE REVIEW

**Review by cmurphy — APPROVED** [review]

File: controller/aks-cluster-config-handler.go [hunk 1]
```diff
@@ -355,10 +355,25 @@ func (h *Handler) validateConfig(config *aksv1.AKSClusterConfig) error {
 		return fmt.Errorf(cannotBeNilError, "azureCredentialSecret", config.ClusterName)
 	}
 
-	if _, err = aks.GetSecrets(h.secretsCache, &config.Spec); err != nil {
+	cred, err := aks.GetSecrets(h.secretsCache, &config.Spec)
+	if err != nil {
 		return fmt.Errorf("couldn't get secret [%s] with error: %v", config.Spec.AzureCredentialSecret, err)
 	}
 
+	cannotBeNilCCError := "field [%s] must be provided for cloud credentials [%s]"
```
Comment (thedadams) [inline controller/aks-cluster-config-handler.go:363]: nit:
```suggestion
	cannotBeEmptyCCError := "field [%s] must be provided for cloud credentials [%s]"
```

**Review by thedadams — COMMENTED** [review]
This method containing these changes is only called on cluster create. Should we consider checking the cloud credential on create, import, and update? It is possible that a user could change the cloud credential on a cluster after it is created.

File: controller/aks-cluster-config-handler.go (hunk 1 above)
Comment (mjura) [reply controller/aks-cluster-config-handler.go]: Fixed

**Comment by cmurphy** [comment]
> This method containing these changes is only called on cluster create. Should we consider checking the cloud credential on create, import, and update? It is possible that a user could change the cloud credential on a cluster after it is created.

@thedadams this method is run before moving to the importing phase so I think it's fine there https://github.com/rancher/aks-operator/blob/ff05137c5e97d4d3e26b7f61bb894c193bc829d2/controller/aks-cluster-config-handler.go#L197-L201

I think the point about validating for updates is valid though, @mjura what do you think?

**Comment by mjura** [comment]
@thedadams @cmurphy yes, it is safe to do this in `validateConfig`, because it will be checked before importing and creating phase. I have tested it and works fine

**Review by rmweir — APPROVED** [review]
LGTM

**Comment by thedadams** [comment]
@mjura It is possible to change the cloud credential on a cluster. If a user creates an AKS cluster with valid cloud credential and then changes the cloud credential on the cluster to an invalid one, then the updates may fail.

Is this a concern we should address?

**Comment by mjura** [comment]
> @mjura It is possible to change the cloud credential on a cluster. If a user creates an AKS cluster with valid cloud credential and then changes the cloud credential on the cluster to an invalid one, then the updates may fail.
> 
> Is this a concern we should address?

We can add `validateConfig` to function during update phase. I will test it

**Comment by mjura** [comment]
It is not needed any more. It is validated in https://github.com/rancher/aks-operator/blob/master/pkg/aks/client.go#L97
and in Rancher master it is validated also by UI.

**Comment by cbron** [comment]
@mjura was there an issue tracking this, if not please create one and send to test, write up how QA can test.

**Pull request closed without merging**